### PR TITLE
Static mem pools

### PR DESCRIPTION
**Pull request by pablogs9**
This PR:
- Refactors incoming buffers to a shared static approach
- Refactor the XML/ref static buffer handling

### CODE REVIEW

**Comment by pablogs9** [comment]
Marie Kondo

**Comment by pablogs9** [comment]
Marie Kondo

File: rmw_microxrcedds_c/test/test_sizes.cpp [hunk 1]
```diff
@@ -37,7 +36,7 @@ TEST_F(RMWBaseTest, estimate_default_static_memory)
     unsigned long subscription_size = sizeof(rmw_uxrce_subscription_t);
     unsigned long publisher_size = sizeof(rmw_uxrce_publisher_t);
     unsigned long node_size = sizeof(rmw_uxrce_node_t);
-    // unsigned long static_input_buffer_size = sizeof(rmw_uxrce_static_input_buffer_t);
+    unsigned long static_input_buffer_size = sizeof(rmw_uxrce_static_input_buffer_t);
 
     fprintf(stderr, "# Static memory analisys \n");
```
Comment (jamoralp) [inline rmw_microxrcedds_c/test/test_sizes.cpp:41]:
```suggestion
    fprintf(stderr, "# Static memory analysis \n");
```

File: rmw_microxrcedds_c/src/rmw_take.c [hunk 2]
```diff
@@ -53,26 +53,27 @@ rmw_take_with_info(
...
 
+    rmw_uxrce_static_input_buffer_t* static_buffer = (rmw_uxrce_static_input_buffer_t*)static_buffer_item->data;
+
     ucdrBuffer temp_buffer;
     ucdr_init_buffer(
-        &temp_buffer, custom_subscription->micro_buffer[custom_subscription->history_read_index],
-        custom_subscription->micro_buffer_lenght[custom_subscription->history_read_index]);
+        &temp_buffer,
+        static_buffer->buffer,
+        static_buffer->lenght);
```
Comment (jamoralp) [inline rmw_microxrcedds_c/src/rmw_take.c:70]:
```suggestion
        static_buffer->length);
```

File: rmw_microxrcedds_c/src/types.h [hunk 3]
```diff
@@ -209,13 +189,24 @@ typedef struct rmw_uxrce_node_t
     uxrObjectId participant_id;
 } rmw_uxrce_node_t;
 
+typedef struct rmw_uxrce_static_input_buffer_t
+{
+    rmw_uxrce_mempool_item_t mem;
+
+    uint8_t buffer[RMW_UXRCE_MAX_INPUT_BUFFER_SIZE];
+    size_t lenght;
```
Comment (jamoralp) [inline rmw_microxrcedds_c/src/types.h:197]:
```suggestion
    size_t length;
```

**Review by jamoralp — CHANGES_REQUESTED** [review]

File: rmw_microxrcedds_c/src/memory.h [hunk 4]
```diff
@@ -26,13 +26,14 @@ typedef struct rmw_uxrce_mempool_item_t
     void*                            data;
 } rmw_uxrce_mempool_item_t;
 
+// #error "this should be locked!"
```
[the remark this answers is not in the document]
Comment (pablogs9) [reply rmw_microxrcedds_c/src/memory.h:29]:
```suggestion
```

**Review by Acuadros95 — APPROVED** [review]

**Pull request merged**